rsocket-core/src/main/java/io/rsocket/metadata/MimeMetadataCodec.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.rsocket.metadata;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.util.CharsetUtil;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A flyweight class that can be used to encode/decode stream data mime types.
+ *
+ * <p>For more on the above metadata formats, see the corresponding <a
+ * href="https://github.com/rsocket/rsocket/blob/master/Extensions/PerStreamDataMimeTypesDefinition.md">Per
+ * stream data mime types definition</a>
+ */
+public class MimeMetadataCodec {
+  static final int STREAM_METADATA_KNOWN_MASK = 0x80; // 1000 0000
+  static final byte STREAM_METADATA_LENGTH_MASK = 0x7F; // 0111 1111
+
+  private MimeMetadataCodec() {}
+
+  /**
+   * made up of a single byte: this represents an encoded mime id, which can be further decoded
+   * using {@link #decode(ByteBuf)}
```

Comment:
```suggestion
   * Encode a {@link WellKnownMimeType} into a newly allocated {@link ByteBuf}.
   * This can then be decoded using {@link #decode(ByteBuf)}.
```